### PR TITLE
Swap deployment to daemonset for node manager

### PR DESCRIPTION
**Pull request by mhmxs**
This change replaces node manager deployment to daemonset, and updates `minAvailable` of PDB instead of `replicas` of the deployment.

Dependant of: https://github.com/storageos/node-manager/pull/55

### CODE REVIEW

**Comment by nolancon** [comment]
@mhmxs is this intended for `v2.10.0` release?

**Review by nolancon — APPROVED** [review]

**Comment by mhmxs** [comment]
> @mhmxs is this intended for `v2.10.0` release?

I don't think so. This feature is not GA, so we aren't rushing :)

File: channels/packages/after-install/0.1.1/pod-disruption-budget.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ kind: PodDisruptionBudget
 metadata:
   name: storageos-node-guard
 spec:
-  maxUnavailable: 1
+  minAvailable: 99999
```
Comment (nolancon) [inline channels/packages/after-install/0.1.1/pod-disruption-budget.yaml:6]: Does it work to specify this as `"100%"` instead? as it says in the api https://github.com/kubernetes/api/blob/a26a16a095cab454e928a95c533b8cf1b80aa2ec/policy/v1/types.go#L31

**Review by nolancon — APPROVED** [review]

File: channels/packages/after-install/0.1.1/pod-disruption-budget.yaml (hunk 1 above)
Comment (mhmxs) [reply channels/packages/after-install/0.1.1/pod-disruption-budget.yaml:6]: @nolancon after install set the right number before creation, so it doesn't matter

**Pull request merged**